src/features/AutoCompletion/model/slice.ts

```diff
@@ -0,0 +1,63 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { getProgressId } from '~/abstract/lib';
+
+type DefaultProps = {
+  entityId: string;
+  eventId: string;
+};
+
+export type SetAutoCompletionPayload = DefaultProps & {
+  autoCompletion: AutoCompletion;
+};
+
+export type SetAutoCompletionStatus = DefaultProps & {
+  status: AutoCompletionStatus;
+};
+
+export type AutoCompletionStatus = 'notStarted' | 'inProgress' | 'completed';
+
+type AutoCompletion = {
+  status: AutoCompletionStatus;
+  lastProcessedActivityId: string | null; // If null, then it means that auto completion is not started and we have no processed activities yet
```

Comment:
You can narrow the type to handle the case you described in the comment:

```
type NotStarted = {
  status: 'notStarted';
  lastProcessedActivityId: null;
};

type InProgressOrCompleted = {
  status: 'inProgress' | 'completed';
  lastProcessedActivityId: string;
};

type AutoCompletion = NotStarted | InProgressOrCompleted;
```

Reply:
Great idea. I have changed it. Thank you.